File: ContactsList_App/app/src/main/java/ma/example/contactlist/SplashActivity.java
package ma.example.contactlist;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        @SuppressLint({"MissingInflatedId", "LocalSuppress"}) ImageView logo = findViewById(R.id.logo);

        logo.setScaleX(0.5f);
        logo.setScaleY(0.5f);
        logo.setAlpha(0f);

        logo.animate()
                .scaleX(1f)
                .scaleY(1f)
                .alpha(1f)
                .setDuration(2000)
                .withEndAction(new Runnable() {
                    @Override
                    public void run() {

                        logo.postDelayed(new Runnable() {
                            @Override
                            public void run() {
                                Intent intent = new Intent(SplashActivity.this, MainActivity.class);
                                startActivity(intent);
                                finish();
                            }
                        }, 1000);
                    }
                })
                .start();
    }

    @Override
    protected void onPause() {
        super.onPause();
        this.finish();
    }
}

File: AgriAlert/AgriAlert_Mobile/app/src/main/java/ma/ensaj/agri_alert/SplashActivity.kt
package ma.ensaj.agri_alert

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.widget.ImageView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.work.Constraints
import androidx.work.ExistingPeriodicWorkPolicy
import androidx.work.NetworkType
import androidx.work.PeriodicWorkRequestBuilder
import androidx.work.WorkManager
import ma.ensaj.agri_alert.service.CropAnalysisService
import ma.ensaj.agri_alert.worker.CropAnalysisWorker
import java.util.concurrent.TimeUnit

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        startCropAnalysisWorker()
        supportActionBar?.hide()
        window.statusBarColor = ContextCompat.getColor(this, R.color.my_dark)

        val logo: ImageView = findViewById(R.id.logo)

        logo.animate()
            .scaleX(0.7f)
            .scaleY(0.7f)
            .setDuration(2000)
            .withEndAction {
                logo.animate()
                    .translationY(-100f)
                    .setDuration(1000)
                    .withEndAction {
                        logo.animate()
                            .alpha(0f)
                            .setDuration(4000)
                            .start()
                    }
                    .start()
            }
            .start()

        val thread = Thread {
            try {
                Thread.sleep(4000)
                val intent = Intent(this@SplashActivity, MainActivity::class.java)
                startActivity(intent)
                finish()
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
        }
        thread.start()
    }

    private fun startCropAnalysisWorker() {
        val constraints = Constraints.Builder()
            .setRequiredNetworkType(NetworkType.CONNECTED)
            .build()

        val workRequest = PeriodicWorkRequestBuilder<CropAnalysisWorker>(
            2, TimeUnit.SECONDS // Repeat every 6 hours
        )
            .setConstraints(constraints)
            .build()

        WorkManager.getInstance(applicationContext).enqueueUniquePeriodicWork(
            "CropAnalysisWorker",
            ExistingPeriodicWorkPolicy.KEEP, // Avoid overwriting existing tasks
            workRequest
        )
    }

    override fun onPause() {
        super.onPause()
        finish()
    }
}
